default/methods/source_control/file/file_browser.py

```diff
@@ -664,59 +704,33 @@ def file_view_core(
-                member = self.member,
-                object_id = self.directory.id,
-                object_type = ValidObjectTypes.dataset,
-                perm = DatasetPermissions.dataset_view
-            )
-            if not perm_result.allowed:
-                self.log['error']['unauthorized'] = f'Cannot view dataset ID: {self.directory.id}'
-                resp = jsonify(log=self.log)
-                resp.status = 401
-                raise Unauthorized(response = resp)
-
-            query, count = WorkingDirFileLink.file_list(
-                session = self.session,
-                working_dir_id = self.directory.id,
-                ann_is_complete = ann_is_complete,
-                type = media_type_query,
-                return_mode = "query",
-                limit = self.metadata["limit"],
-                date_from = self.metadata["date_from"],
-                date_to = self.metadata["date_to"],
-                issues_filter = self.metadata["issues_filter"],
-                offset = self.metadata["start_index"],
-                original_filename = self.metadata['search_term'],
-                order_by_class_and_attribute = File.id,
-                order_by_direction = order_by_direction,
-                exclude_removed = exclude_removed,
-                file_view_mode = self.metadata['file_view_mode'],
-                job_id = job_id,
-                has_some_machine_made_instances = has_some_machine_made_instances,
-                ignore_id_list = ignore_id_list,
-                count_before_limit = True,
-                include_children_compound = self.metadata['with_children_files']
-
-            )
-
-            file_count += count
-
-            working_dir_file_list = query.all()
+        query, count = WorkingDirFileLink.file_list(
+            session = self.session,
+            working_dir_id = self.directory.id,
+            ann_is_complete = ann_is_complete,
+            type = media_type_query,
+            return_mode = "query",
+            limit = self.metadata["limit"],
+            date_from = self.metadata["date_from"],
+            date_to = self.metadata["date_to"],
+            issues_filter = self.metadata["issues_filter"],
+            offset = self.metadata["start_index"],
+            original_filename = self.metadata['search_term'],
+            order_by_class_and_attribute = File.id,
+            order_by_direction = order_by_direction,
+            exclude_removed = exclude_removed,
+            file_view_mode = self.metadata['file_view_mode'],
+            job_id = job_id,
+            has_some_machine_made_instances = has_some_machine_made_instances,
+            ignore_id_list = ignore_id_list,
+            count_before_limit = True,
+            include_children_compound = self.metadata['with_children_files']
+        )
 
+        working_dir_file_list = query.all()
+        return working_dir_file_list, count
 
+    def add_pagination_metadata(self, file_count):
```

Comment:
`add_pagination_metadata` replaces [this block](https://github.com/diffgram/diffgram/blob/38d12c685c47a53061bcbe9b80c1ae257e651a84/default/methods/source_control/file/file_browser.py#L720-L729) of code